test/algorithms/test_qft.cpp

```diff
@@ -1,40 +1,25 @@
 #include "algorithms/QFT.hpp"
+#include "dd/Benchmark.hpp"
 #include "dd/FunctionalityConstruction.hpp"
-#include "dd/Simulation.hpp"
 
 #include "gtest/gtest.h"
 #include <cmath>
 #include <iostream>
 
 class QFT : public testing::TestWithParam<std::size_t> {
 protected:
-  void TearDown() override {
-    dd->decRef(sim);
-    dd->decRef(func);
-    dd->garbageCollect(true);
-
-    // number of complex table entries after clean-up should equal initial
-    // number of entries
-    EXPECT_EQ(dd->cn.realCount(), initialComplexCount);
-    // number of available cache entries after clean-up should equal initial
-    // number of entries
-    EXPECT_EQ(dd->cn.cacheCount(), initialCacheCount);
-  }
+  void TearDown() override { dd->garbageCollect(true); }
 
   void SetUp() override {
     nqubits = GetParam();
-    dd = std::make_unique<dd::Package<>>(nqubits);
-    initialCacheCount = dd->cn.cacheCount();
-    initialComplexCount = dd->cn.realCount();
+    exp = std::make_unique<dd::Experiment>();
+    exp->dd = std::make_unique<dd::Package<>>(nqubits);
   }
 
   std::size_t nqubits = 0;
-  std::unique_ptr<dd::Package<>> dd;
   std::unique_ptr<qc::QFT> qc;
-  std::size_t initialCacheCount = 0;
-  std::size_t initialComplexCount = 0;
-  qc::VectorDD sim{};
-  qc::MatrixDD func{};
+  std::unique_ptr<dd::Package<>> dd;
```

Comment:
I do not see how this member should be necessary as the experiment inherently contains a package instance that is adequately sized.